File: Data Structures & Algorithms/find-duplicate-integer/submission-0.java
class Solution {
    public int findDuplicate(int[] nums) {
        Map<Integer, Integer> map = new HashMap<>();
        for (int i = 0; i < nums.length; ++i) {
            if (map.get(nums[i]) != null) {
                return nums[i];
            }
            map.put(nums[i], 1);
        }
        return -1;
    }
}

File: Data Structures & Algorithms/two-integer-sum/submission-0.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        table = {}
        for i in range(len(nums)):
            diff = target - nums[i]
            if table.get(nums[i], None) != None:
                # print(table.get(nums[i]), i)
                return list([table[nums[i]], i])
            else:
                table[diff] = i
        # print(123)
        # table = {4:0, , }
        return list()